ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/ReadSymbolsFromMobileStyle/ReadSymbolsFromMobileStyle.cpp

```diff
@@ -0,0 +1,255 @@
+// [WriteFile Name=ReadSymbolsFromMobileStyle, Category=DisplayInformation]
+// [Legal]
+// Copyright 2019 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#include "ReadSymbolsFromMobileStyle.h"
+
+#include "Map.h"
+#include "MapQuickView.h"
+#include "GraphicsOverlay.h"
+#include "Graphic.h"
+#include "SymbolStyle.h"
+#include "MultilayerSymbol.h"
+#include "Point.h"
+#include "MultilayerPointSymbol.h"
+
+#include "SymbolImageProvider.h"
+
+#include <QDir>
+#include <QObject>
+#include <QQmlContext>
+#include <QTemporaryDir>
+#include <QtCore/qglobal.h>
+
+#ifdef Q_OS_IOS
+#include <QStandardPaths>
+#endif // Q_OS_IOS
+
+using namespace Esri::ArcGISRuntime;
+
+// helper method to get cross platform data path
+namespace {
+QString defaultDataPath()
+{
+  QString dataPath;
+
+#ifdef Q_OS_ANDROID
+  dataPath = "/sdcard";
+#elif defined Q_OS_IOS
+  dataPath = QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation);
+#else
+  dataPath = QDir::homePath();
+#endif
+
+  return dataPath;
+}
+} // namespace
```

Comment:
At some point perhaps we need a set of common utils for the samples app. This block of code keeps popping up.

Reply:
yeah, the struggle is how do we keep sample projects atomic examples that don't have outside dependencies. We were pulling in Extras before, but we felt that was adding unnecessary dependency

Reply:
yeah good point. This is fine.